apps/meteor/app/emoji-custom/server/lib/uploadEmojiCustom.ts

```diff
@@ -67,8 +70,10 @@ export async function uploadEmojiCustomWithBuffer(
 			encodeURIComponent(`${emojiData.name}.${emojiData.extension}`),
 			contentType,
 		);
-		ws.on('end', () => {
-			setTimeout(() => api.broadcast('emoji.updateCustom', emojiData), 500);
+		ws.on('end', async () => {
+			const etag = Random.hexString(6);
+			await EmojiCustom.setETagByName(emojiData.name, etag);
```

Comment:
@abhinavkrin i did this here but i'm not sure how safe it is to update by name. Are emoji names unique somehow? Will update this potentially update also other emojis?

(Had to fallback to this because the `uploadEmojiCustom` method was only accepting name/alias/file, so making _id required could break it.

Reply:
The emoji names are indeed unique.